### PR TITLE
Refactor to match recommended ESLint setup

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Refactor ESLint setup to match the recommended example from Turborepo: https://github.com/vercel/turbo/tree/main/examples/basic.

Add new workspace `eslint-config-custom` that contains the shared ESLint config. Remove the `scripts` workspace.

Use yarn instead of npx in pre-commit (consistency).

Update imports in `ui` workspace that seems to have been broken since before.

## Justify why they are needed

I was having issues with `lint-staged` not ordering my imports according to our rules. The new setup solves this issue by adding a `.eslintrc.js` config in the root of the project that is used by `lint-staged`.

The recommended way to setup ESLint with Turborepo has also changes since we set it up.

I personally like this setup better since it's easier to extend the shared config compared to before.

There is also a `eslint-config-turbo` but it was causing a lot of errors so let's introduce it separately if at all.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1362** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1362" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1362?utm_source=stack-comment).

File: packages/ui/package.json [hunk 1]
```diff
@@ -36,8 +36,8 @@
     "@types/react-dom": "18.0.10",
     "babel-jest": "29.3.1",
     "babel-loader": "9.1.2",
+    "eslint-config-custom": "workspace:",
```
Comment (alebedev) [inline packages/ui/package.json:39]: Nit: would be nice to add lint script to ui too

**Review by alebedev — APPROVED** [review]
Looks like the right way to go

Let's see what other changes are needed to make it build in CI

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+  root: true,
+  extends: ['custom'],
```
[the remark this answers is not in the document]
Comment (robinandeer) [reply .eslintrc.js:3]: haha I like the official way to extend better

I guess the convention is "custom" comes from a package called `eslint-config-custom`

**Review by KarnellSchultz — APPROVED** [review]
i-win-square-peg-in-a-round-hole-gif

**Pull request merged**